lib/src/config/route/app_routes.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/cupertino.dart';
+import '../../core/util/constants.dart' as constants;
+import '../../domain/repository/i_movie_repository.dart';
+import '../../domain/usecase/implementation/fetch_movies_by_genre_usecase.dart';
+import '../../domain/usecase/implementation/fetch_movies_from_endpoint_usecase.dart';
+import '../../presentation/bloc/movie_from_endpoint_bloc.dart';
+import '../../presentation/bloc/movie_with_genre_bloc.dart';
+import '../../presentation/view/movies_by_genres_view.dart';
+import '../../presentation/view/movies_view.dart';
+
+class AppRoutes {
+  final IMovieRepository repository;
+
+  AppRoutes(this.repository);
+
+  Map<String, Widget Function(BuildContext)>? endpointRoutes;
+
+  Map<String, Widget Function(BuildContext)> get routes {
+    final map = <String, Widget Function(BuildContext)>{};
+    map.addAll(endpointRoutes!);
+    map.addAll(genreRoute);
+    return map;
+  }
+
+  Map<String, Widget Function(BuildContext)> get genreRoute => {
+        constants.genreAppRoute: (context) => MoviesByGenreView(
+              movieWithGenreBloc: MovieWithGenreBloc(
+                fetchMoviesByGenreUsecase: FetchMoviesByGenreUsecase(
+                  repository: repository,
+                ),
+              ),
+            ),
+      };
+
+  Map<String, Widget Function(BuildContext)> getMovieEndpointRoutes(
+    PageController pageController,
+  ) =>
+      endpointRoutes ??= {
```

Comment:
Use the constants defined below for these routes

Reply:
fixed